README.md

```
@@ -101,6 +101,26 @@ It is encouraged that you develop an initial prototype during the application ph
+MiniChain supports fully-functional smart contracts written directly in Python! 
+The execution engine uses `sys.settrace` for precise **Gas Metering** (charging 1 gas per executed opcode) and `multiprocessing` for **Sandboxed Execution** to ensure network security.
+
+### Writing a Contract
+Smart contracts in MiniChain have access to a persistent `storage` dictionary and a `msg` dictionary containing transaction context (`sender`, `value`, `data`).
+
+Check out the `/examples` directory for tutorials:
+- `examples/counter.py` - A basic state mutation example.
+- `examples/stablecoin.py` - A minimal ERC-20 style fungible token.
+- `examples/dex.py` - An Automated Market Maker (AMM) using the constant product formula (x * y = k).
+
+### Interacting via CLI
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)